operator/pkg/helmreconciler/rendering.go

```diff
@@ -175,19 +190,77 @@ func unmarshalAndValidateIOPSpec(iopsYAML string) (*v1alpha1.IstioOperatorSpec,
 func (h *HelmReconciler) ProcessManifest(manifest manifest.Manifest) (int, error) {
 	var errs []error
 	log.Infof("Processing resources from manifest: %s", manifest.Name)
-	objects, err := object.ParseK8sObjectsFromYAMLManifest(manifest.Content)
+	allObjects, err := object.ParseK8sObjectsFromYAMLManifest(manifest.Content)
 	if err != nil {
 		return 0, err
 	}
-	for _, obj := range objects {
+
+	name := h.instance.Name
+
+	objectCachesMu.Lock()
+
+	// Create and/or get the cache corresponding to the CR name we're processing. Per name partitioning is required to
+	// prune the cache to remove any objects not in the manifest generated for a given CR.
+	if objectCaches[name] == nil {
+		objectCaches[name] = &ObjectCache{
+			cache: make(map[string]*object.K8sObject),
+			mu:    &sync.RWMutex{},
+		}
+	}
+	objectCache := objectCaches[name]
+
+	objectCachesMu.Unlock()
+
+	// Ensure that for a given CR name only one control loop uses the per-name cache at any time.
+	objectCache.mu.Lock()
+	defer objectCache.mu.Unlock()
```

Comment:
If I read this correctly, we're effectively holding onto the cache lock for this entire call, which means the threads invoking `ProcessManifest()` become serialized through this function, negating any benefit to splitting the processing up across multiple threads.

Reply:
objectCachesLock (plural) is global and that's held only briefly to get the objectCache lock, which is per-component, so components are not serialized. what is serialized is if two control loops cannot work on the same component for the same CR (as might happen if a reconcile was still in progress and another update came in). that's something that we should really be doing anyway regardless of caching, since i'm not sure if we can rely on eventual consistency to work in all such cases.